codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -545,27 +488,11 @@ void incrementStreamableBytesForTree(int numBytes) {
          * A wrapper class around the content of a data frame.
          */
         private final class Frame {
-            final ByteBuf data;
-            final boolean endStream;
-            final SimplePromiseAggregator promiseAggregator;
-            final ChannelPromise promise;
-            int padding;
+            final FlowControlledPayload payload;
             boolean enqueued;
 
-            Frame(SimplePromiseAggregator promiseAggregator, ByteBuf data, int padding, boolean endStream) {
-                this.data = data;
-                this.padding = padding;
-                this.endStream = endStream;
-                this.promiseAggregator = promiseAggregator;
-                promise = ctx.newPromise();
-                promiseAggregator.add(promise);
-            }
-
-            /**
-             * Gets the total size (in bytes) of this frame including the data and padding.
-             */
-            int size() {
-                return data.readableBytes() + padding;
+            Frame(FlowControlledPayload payload) {
+                this.payload = payload;
             }
 
             void enqueue() {
```

Comment:
Now that we are always queuing.  Should we make this private, call it in the constructor, and remove the `enqueued` member?